src/actions/receive-profile.js

```diff
@@ -295,10 +293,13 @@ export function finalizeFullProfileView(
       hiddenLocalTracksByPid = newHiddenTracksByPid;
     }
 
-    selectedThreadIndexes = initializeSelectedThreadIndex(
-      selectedThreadIndexes,
-      visibleThreadIndexes,
-      profile
+    // Wrap selectedThreadIndexes with ensureExists to pass the CI flow-checks
+    selectedThreadIndexes = ensureExists(
+      initializeSelectedThreadIndex(
+        selectedThreadIndexes,
+        visibleThreadIndexes,
+        profile
+      )
```

Comment:
The problem is that currently `selectedThreadIndexes` is a parameter, so it could be changed over time. Probably previously it was accepting the return value of `initializeSelectedThreadIndex` because this was straightforward, but now when using it inside a callback of `withHistoryReplaceStateSync` it takes the initial type instead.

The solution, I think, is that you need to set it to another variable that would be `const`. I think I'd change the name of the parameter too.
```js
export function finalizeFullProfileView(
  profile: Profile,
  selectedThreadIndexesOrNull: Set<ThreadIndex> | null
) {
...

const selectedThreadIndexes = initializeSelectedThreadIndex(
      selectedThreadIndexesOrNull,
      visibleThreadIndexes,
      profile
);

...
```

`ensureExists` should be unnecessary here, because `initializeSelectedThreadIndex` always return a non-null value.

Hope this makes sense!